common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp

```diff
@@ -474,6 +474,12 @@ bool ap_wlan_hal_nl80211::read_supported_channels()
     return true;
 }
 
+bool ap_wlan_hal_nl80211::update_preference_channels_from_supported_channels()
+{
+    m_radio_info.preferred_channels = m_radio_info.supported_channels;
```

Comment:
So you already do this here - so the API is very simple.. Just use `m_radio_info.supported_channels` if ACS is disabled.